public/javascripts/SVLabel/src/SVLabel/mission/MissionProgress.js

```diff
@@ -49,6 +49,39 @@ function MissionProgress (svl, gameEffectModel, missionModel, modalModel, neighb
         );
         mission.complete();
 
+        // mTurk survey prompt, replace mTurk variable with actual assessment of if modal should display
+        // query if 1. user is mturk 2. user has just completed all missions in their *first* HIT
+
+            var url = '/survey/display';
+            var numMissionsBeforeSurvey = 2;
+            var numMissionsCompleted = svl.missionContainer.getCompletedMissions().length;
+            $.ajax({
+                async: true,
+                url: url,//endpoint that checks above conditions
+                type: 'get',
+                success: function(data){
+                    if(data.displayModal && numMissionsCompleted == numMissionsBeforeSurvey){
```

Comment:
why is the condition -- numMissionsCompleted == numMissionsBeforeSurvey within the ajax call? Shouldn't the call be made if numMissionsCompleted == numMissionsBeforeSurvey?

Reply:
Originally the survey/display endpoint relied on information provided by the survey controller to display the survey. This included the count of the number of missions. The survey should only trigger based on the information returned by the ajax call. But now, the mission count is coming from the mission container which isnt ideal. Like I've mentioned in my comment below. The updation of the mission_user table happens asynchronously which messes with the calculated count of the number of completed missions. I'm leaving the ajax call here just in case we have a different way to trigger survey events in the future (every 3rd event etc)..